Cargo.toml

```diff
@@ -5,6 +5,8 @@ members = [
   "crates/xmtp-keystore",
   "crates/corecrypto",
   "crates/xmtp-networking",
+  "crates/libxmtp-core",
```

Comment:
Is it worth tagging with `v3`? Or is libxmtp-core's scope going to go beyond? I do think once we pull in vodozemac we might hit conflicts that require excluding it from the Cargo workspace, but we can cross that bridge when we get there.

Reply:
It's a good question, I'm not sure if consumers will even see this crate directly. They might consume an intermediary crate (like a bindings/wasm crate). In that sense we have a lot of flexibility with naming to make things clearer. `libxmtp-core` is totally fine since it'll hopefully be very clear what code lives in it (V3 for now + some state management goodies maybe)